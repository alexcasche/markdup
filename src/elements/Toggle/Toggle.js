import styled from 'styled-components'
import ReactToggle from 'react-toggle'

const Toggle = styled(ReactToggle)`
  display: flex;
  align-items: center;
  .react-toggle {
    .react-toggle-thumb {
      box-shadow: none !important;
    }
  }
`

export default Toggle
