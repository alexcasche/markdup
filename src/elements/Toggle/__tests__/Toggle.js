import Toggle from '../Toggle'

describe('Toggle Element', () => {
  it('renders correctly', () => {
    const tree = render(<Toggle />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
