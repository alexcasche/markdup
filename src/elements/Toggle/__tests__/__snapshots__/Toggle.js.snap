// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .react-toggle .react-toggle-thumb {
  box-shadow: none !important;
}

<div
  className="react-toggle c0"
  onClick={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="react-toggle-track"
  >
    <div
      className="react-toggle-track-check"
    >
      <svg
        height="11"
        viewBox="0 0 14 11"
        width="14"
      >
        <title>
          switch-check
        </title>
        <path
          d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
          fill="#fff"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="react-toggle-track-x"
    >
      <svg
        height="10"
        viewBox="0 0 10 10"
        width="10"
      >
        <title>
          switch-x
        </title>
        <path
          d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
          fill="#fff"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    className="react-toggle-thumb"
  />
  <input
    className="react-toggle-screenreader-only"
    onBlur={[Function]}
    onFocus={[Function]}
    type="checkbox"
  />
</div>
`;
